cpp/src/parquet/page_index_benchmark.cc

```diff
@@ -0,0 +1,153 @@
+      data[i] = d(gen);
+    }
+  } else if constexpr (std::is_floating_point_v<T>) {
+    std::default_random_engine gen(seed);
+    std::uniform_real_distribution<T> d(std::numeric_limits<T>::lowest(),
+                                        std::numeric_limits<T>::max());
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i] = d(gen);
+    }
+  } else if constexpr (std::is_same_v<FLBA, T>) {
+    GenerateRandomString(kDataStringLength * size, seed, heap);
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i].ptr = heap->data() + i * kDataStringLength;
+    }
+  } else if constexpr (std::is_same_v<ByteArray, T>) {
+    GenerateRandomString(kDataStringLength * size, seed, heap);
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i].ptr = heap->data() + i * kDataStringLength;
+      data[i].len = kDataStringLength;
+    }
+  } else if constexpr (std::is_same_v<Int96, T>) {
+    std::default_random_engine gen(seed);
+    std::uniform_int_distribution<int> d(std::numeric_limits<int>::min(),
+                                         std::numeric_limits<int>::max());
+    for (uint32_t i = 0; i < size; ++i) {
+      data[i].value[0] = d(gen);
+      data[i].value[1] = d(gen);
+      data[i].value[2] = d(gen);
+    }
+  }
+}
+
+template <typename DType>
+void BM_ReadColumnIndex(::benchmark::State& state) {
+  schema::NodePtr type = ::parquet::schema::PrimitiveNode::Make(
+      "b", Repetition::OPTIONAL, DType::type_num, ConvertedType::NONE, 8);
+  auto descr_ptr =
+      std::make_unique<ColumnDescriptor>(type, /*def_level=*/1, /*rep_level=*/0);
+  auto descr = descr_ptr.get();
+
+  const size_t num_pages = state.range(0);
+
+  auto builder = ColumnIndexBuilder::Make(descr);
+
+  const size_t values_per_page = 100;
+  for (int i = 0; i < num_pages; ++i) {
+    auto stats = MakeStatistics<DType>(descr);
+    std::vector<uint8_t> heap;
+    std::vector<typename DType::c_type> values;
+    values.resize(values_per_page);
+    GenerateBenchmarkData(values_per_page, /*seed=*/0, values.data(), &heap);
+    stats->Update(values.data(), values_per_page, 0);
+    builder->AddPage(stats->Encode());
+  }
+
+  builder->Finish();
+  auto sink = CreateOutputStream();
+  builder->WriteTo(sink.get());
+  auto buffer = sink->Finish().ValueOrDie();
+  ReaderProperties properties;
+  for (auto _ : state) {
+    auto column_index =
+        ColumnIndex::Make(*descr, buffer->data() + 0, buffer->size(), properties);
+    ::benchmark::DoNotOptimize(column_index);
+  }
+  state.counters["num_pages"] = num_pages;
+}
+
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, Int32Type)->Range(8, 1024);
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, Int64Type)->Range(8, 1024);
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, FloatType)->Range(8, 1024);
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, DoubleType)->Range(8, 1024);
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, FLBAType)->Range(8, 1024);
+BENCHMARK_TEMPLATE(BM_ReadColumnIndex, ByteArrayType)->Range(8, 1024);
```

Comment:
Hmm, why so many types? We don't expect significant differences between Int32 and Int64, or Float and Double, I think. Therefore we can probably reduce it to: Int64, Double, FLBA, ByteArray.